pkg/cli/userfile.go

```diff
@@ -69,8 +104,117 @@ func openUserFile(source string) (io.ReadCloser, error) {
 	return f, nil
 }
 
+// Construct the userfile ExternalStorage URI from CLI args.
+func constructUserfileDestinationURI(source, destination, user string) string {
+	// User has not specified a destination URI/path. We use the default URI
+	// scheme and host, and the basename from the source arg as the path.
+	if destination == "" {
+		sourceFilename := filepath.Base(source)
```

Comment:
this should be `path` not `filepath` since it is a host-independent network supplied path.